src/Constraint/Contains.php

```diff
@@ -64,7 +64,7 @@ public function evaluate($value)
     public function equivalentTo(Expression $other)
     {
         // Since this class is final, we can check with instanceof
-        return $other instanceof $this && $this->comparedValue == $other->comparedValue;
+        return $other instanceof $this && (string) $this->comparedValue === (string) $other->comparedValue;
```

Comment:
Avoiding it is not possible, since then the logic of the method is broken. Doing so in the constructor is a good idea though.